### PR TITLE
Obmin 353 redesign pop up of categories

### PR DESCRIPTION
**Pull request by Kyslytsia**
Redesign pop up of categories

### CODE REVIEW

File: src/components/category-button/CategoryButton.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  title: 'CategoryButton',
+  component: CategoryButton,
+  argTypes,
+} satisfies Meta<typeof CategoryButton>;
+
+export default meta;
+type Story = StoryObj<typeof CategoryButton>;
+
+const Template = (args: any) => {
+  return (
```
Comment (ASGladchenko) [inline src/components/category-button/CategoryButton.stories.tsx:19]: mb need change types any on correct type

File: src/components/category-button/types.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+export interface ICategoryInfo {
+  link: string;
+  img: string;
+  text: string;
+}
```
Comment (ASGladchenko) [inline src/components/category-button/types.ts:5]: sort

**Review by ASGladchenko — APPROVED** [review]

**Review by Voloshkina — APPROVED** [review]

**Review by Viktoria-Serkiz — APPROVED** [review]

File: src/components/category-button/index.tsx [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { Responsive } from 'components';
-import { NavCategory } from 'components/category-nav';
+import { NavCategory } from 'components/category-button/category-nav';
 import { useDelayAnimation } from 'hooks/useDelayAnimation';
```
Comment (cloudmenson) [inline src/components/category-button/index.tsx:4]: sort

File: src/components/category-button/index.tsx [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { Responsive } from 'components';
+import { useDelayAnimation } from 'hooks/useDelayAnimation';
+import { NavCategory } from 'components/category-button/category-nav';
```
Comment (Wolshebnik) [inline src/components/category-button/index.tsx:3]: if component exist in the same folder
then import from './category-nav'

File: src/components/category-button/category-nav/index.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+
+import { images } from './config';
+import * as Styles from './styles';
+import { ICategoryNav } from './types';
+import { ICategoryInfo } from '../types';
+
+const NavCategory = ({ categoryInfo = [], isOpen, delay }: ICategoryNav) => {
+  const [isVisibleSun, setVisibleSun] = useState<boolean>(true);
+
+  const isLeave = true;
```
Comment (Wolshebnik) [inline src/components/category-button/category-nav/index.tsx:13]: remove

File: src/components/category-button/category-nav/index.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+                    <Styles.SunMain
+                      alt="sun-main"
+                      variant={el.text}
+                      src={Images.sunMain}
+                    />
+                  )}
+
+                  <Styles.SunCategory
+                    src={sun}
+                    isLeave={isLeave}
```
Comment (Wolshebnik) [inline src/components/category-button/category-nav/index.tsx:48]: just isLeave and I don't understand why you use isLeave
It's always true

File: src/components/category-button/styles.ts [hunk 7]
```diff
@@ -0,0 +1,66 @@
+import styled, { css } from 'styled-components';
+
+export const CategoriesText = styled.div`
+  text-transform: uppercase;
+
+  ${({ theme }) => css`
+    color: ${theme.colors.categoryBtn.colorText};
```
Comment (Wolshebnik) [inline src/components/category-button/styles.ts:7]: If just one param 
color: ${ ({theme}) => theme.colors.categoryBtn.colorText}

File: src/components/category-button/styles.ts [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+  `}
+`;
+
+export const CategoriesDesktop = styled.div<{
+  isOpen: boolean;
+}>`
+  position: relative;
+  box-sizing: border-box;
```
Comment (Wolshebnik) [inline src/components/category-button/styles.ts:15]: this param must be below

File: src/components/category-button/styles.ts [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+    css`
+      &:after {
+        transform: rotate(-180deg);
+      }
+    `}
+  `}
+`;
+
+export const Categories = styled.div`
+  box-sizing: border-box;
```
Comment (Wolshebnik) [inline src/components/category-button/styles.ts:53]: the same

File: src/components/category-button/types.ts [hunk 10]
```diff
@@ -0,0 +1,12 @@
+export interface ICategoryInfo {
+  link: string;
+  img?: string;
+  sun?: string;
```
Comment (Wolshebnik) [inline src/components/category-button/types.ts:4]: params img, sun no need anymore

File: src/components/category-button/CategoryButton.stories.tsx [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    <div
+      style={{
+        height: '1500px',
+        backgroundColor: '#53b2d4',
+      }}
+    >
+      <PresentationHeader>
+        <CategoryButton {...args}></CategoryButton>
+      </PresentationHeader>
```
Comment (cloudmenson) [inline src/components/category-button/CategoryButton.stories.tsx:29]: <CategoryButton {...args} .>

File: src/components/category-button/styles.ts [hunk 12]
```diff
@@ -0,0 +1,66 @@
...
+
+    &:after {
+      content: '';
+      position: absolute;
+      right: 25px;
+      bottom: 20px;
+      border: 0 solid transparent;
+      border-right-width: 9px;
+      border-left-width: 9px;
+      border-bottom: 8px solid ${theme.colors.categoryBtn.colorText};
```
Comment (cloudmenson) [inline src/components/category-button/styles.ts:39]: border-width: 0 9px;

File: src/components/category-button/styles.ts [hunk 13]
```diff
@@ -0,0 +1,66 @@
...
+  align-items: center;
+  cursor: pointer;
+
+  ${({ theme }) => css`
+    ${theme.responsive.isMobile &&
+    css`
+      height: 30px;
+      width: 30px;
+    `}
+  `}
```
Comment (cloudmenson) [inline src/components/category-button/styles.ts:65]: ${({ theme }) =>
  theme.responsive.isMobile &&
  css`
    height: 30px;
    width: 30px;
  `}

File: src/hooks/useDelayAnimation.ts [hunk 14]
```diff
@@ -9,7 +9,11 @@ export const useDelayAnimation = (delay: number) => {
 
     if (open) {
       setIsOpen(true);
-      setTimeout(() => setIsAnimation(true), 1);
+      if (!immediateStart) {
+        setTimeout(() => setIsAnimation(true), 1);
+      } else {
+        setIsAnimation(true);
+      }
```
Comment (cloudmenson) [inline src/hooks/useDelayAnimation.ts:16]: if (open) {
  setIsOpen(true);
  if (!immediateStart) setTimeout(() => setIsAnimation(true), 1);
  else setIsAnimation(true);
}

File: src/components/category-button/index.tsx [hunk 15]
```diff
@@ -0,0 +1,44 @@
+import { Responsive } from 'components';
+import { NavCategory } from './category-nav';
```
Comment (Wolshebnik) [inline src/components/category-button/index.tsx:2]: wrong sort

**Review by Wolshebnik — APPROVED** [review]

**Review by cloudmenson — APPROVED** [review]

**Review by Mish-hash — APPROVED** [review]

**Pull request merged**